server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -422,6 +422,68 @@ public void testDeleteOperations() throws Exception {
         }
     }
 
+    /**
+     * This tests that the max seqNo we send to replicas is accurate and that after failover
+     * the new primary starts indexing from the correct maxSeqNo and replays the correct count of docs
+     * from xlog.
+     */
+    public void testReplicationPostDeleteAndForceMerge() throws Exception {
```

Comment:
Thank you @mch2 for this quick fix.
1. I am able to have this test pass successfully in my local without other changes. Is it possible to update this test so that it fails without your changes.
2. Not related to this test. Can we add unit tests that verify behaviour of seqNo with deletes & force merges ?

Reply:
1 - hmm this is failing for me if i copy/paste to main as is with
```
三月 09, 2023 5:31:40 下午 com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
警告: Uncaught exception in thread: Thread[#83,opensearch[node_t0][flush][T#1],5,TGRP-SegmentReplicationIT]
java.lang.AssertionError
	at __randomizedtesting.SeedInfo.seed([BC702393CD92400E]:0)
	at org.opensearch.index.engine.Engine.getMaxSeqNoFromSearcher(Engine.java:317)
	at org.opensearch.index.engine.Engine.getMaxSeqNoFromSegmentInfos(Engine.java:291)
	at org.opensearch.index.shard.IndexShard.lambda$getLatestSegmentInfosAndCheckpoint$7(IndexShard.java:1492)
```.

2. Hmm this issue is happening because we were previously invoking `engine.getMaxSeqNoFromSegmentInfos` from the latest SegmentInfos.  I don't see any issue/change with that method's functionality, only that we were using it when we shouldn't have.  So I think this IT better tests for this case?

Reply:
1. Sorry, i was not clear above. I was testing in `2.x` branch, i see it passes. I verified it fails on `main` with similar assertion as you pasted above.
2. Yes, make sense.